src/tls/keypair.h

```diff
@@ -751,13 +833,87 @@ namespace tls
       return {crt->raw.p, crt->raw.p + crt->raw.len};
     }
 
-    ~Verifier()
+    virtual ~Verifier()
     {
       mbedtls_x509_crt_free(&cert);
-#if CURVE_CHOICE_SECP256K1_BITCOIN
-      if (ctx)
-        secp256k1_context_destroy(ctx);
-#endif
     }
   };
+
+  class Verifier_k1Bitcoin : public Verifier
+  {
+  protected:
+    secp256k1_context* k1_ctx =
+      secp256k1_context_create(SECP256K1_CONTEXT_VERIFY);
+    secp256k1_pubkey c4_pub;
+
+  public:
+    template <typename... Ts>
+    Verifier_k1Bitcoin(Ts... ts) : Verifier(std::forward<Ts>(ts)...)
+    {
+      auto k = mbedtls_pk_ec(cert.pk);
+      size_t pub_len;
+      uint8_t pub_buf[100];
```

Comment:
Where does 100 come from?

Reply:
Line 650 in the previous version. There's nothing obvious in the mbedtls documentation giving a suggested size here. I'll give the magic constant a name though.

Reply:
And since this is the same initialisation needed for `PublicKey_k1Bitcoin`, I've extracted it to `parse_secp256k_bc`